huntsman/camera/__init__.py

```diff
@@ -0,0 +1,171 @@
+            n_cameras = len(camera_uris)
+            if n_cameras > 0:
+                msg = "Found {} distributed cameras on name server".format(n_cameras)
+                logger.debug(msg)
+            else:
+                msg = "Found name server but no distributed cameras"
+                logger.warning(msg)
+    except Pyro4.errors.NamingError as err:
+        msg = "Couldn't connect to Pyro name server: {}".format(err)
+        logger.warning(msg)
+        camera_uris = OrderedDict()
+
+    return camera_uris
+
+
+def create_cameras_from_config(config=None, logger=None, **kwargs):
+    """Create camera object(s) based on the config.
+
+    Creates a camera for each camera item listed in the config. Ensures the
+    appropriate camera module is loaded.
+
+    Args:
+        **kwargs (dict): Can pass a `cameras` object that overrides the info in
+            the configuration file. Can also pass `auto_detect`(bool) to try and
+            automatically discover the ports.
+
+    Returns:
+        OrderedDict: An ordered dictionary of created camera objects, with the
+            camera name as key and camera instance as value. Returns an empty
+            OrderedDict if there is no camera configuration items.
+
+    Raises:
+        error.CameraNotFound: Raised if camera cannot be found at specified port or if
+            auto_detect=True and no cameras are found.
+        error.PanError: Description
+    """
+    if not logger:
+        logger = logger_module.get_root_logger()
+
+    if not config:
+        config = load_config(**kwargs)
+
+    # Helper method to first check kwargs then config
+    def kwargs_or_config(item, default=None):
+        return kwargs.get(item, config.get(item, default))
+
+    camera_info = kwargs_or_config('cameras')
+    if not camera_info:
+        logger.info('No camera information in config.')
+        return OrderedDict()
+
+    try:
+        cameras = create_local_cameras(config=config, logger=logger, **kwargs)
+    except (error.PanError, KeyError, error.CameraNotFound):
+        logger.debug("No local cameras")
+        cameras = OrderedDict()
+
+    a_simulator = 'camera' in kwargs_or_config('simulator', default=list())
+    distributed_cameras = kwargs.get('distributed_cameras',
+                                     camera_info.get('distributed_cameras', False))
+    if not a_simulator and distributed_cameras:
+        logger.debug("Creating distributed cameras")
+        cameras.update(create_distributed_cameras(camera_info, logger=logger))
+
+    if len(cameras) == 0:
+        raise error.CameraNotFound(
+            msg="No cameras available. Exiting.", exit=True)
+
+    # Find primary camera
+    primary_camera = None
+    for camera in cameras.values():
+        if camera.is_primary:
+            primary_camera = camera
+
+    # If no camera was specified as primary use the first
+    if primary_camera is None:
+        camera_names = sorted(cameras.keys())
```

Comment:
Minor point, but since it's an OrderedDict do you want to skip the `sorted` and use the first one placed in the dict? I think it comes out the same anyway.

Reply:
I guess it depends on whether we want to default to the first camera added (which will be `Cam00` if there are any local cameras) or to the first camera alphabetically...